tactics/class_tactics.mli

```diff
@@ -41,8 +50,8 @@ val autoapply : constr -> Hints.hint_db_name -> unit Proofview.tactic
 
 module Search : sig
   val eauto_tac :
-    ?st:TransparentState.t
-    (** The transparent_state used when working with local hypotheses  *)
+    Hints.hint_mode array list GlobRef.Map.t * TransparentState.t
+    (** The transparent_state and modes used when working with local hypotheses  *)
```

Comment:
I have wondered before if this function should be removed from the API since it's not used elsewhere. I'm still wondering.

Reply:
I've been wondering too, but I'd rather remove it as part of a specific PR.